modules/standard/Path.chpl

```diff
@@ -368,6 +368,34 @@ proc dirname(name: string): string {
   return dirname(path=name);
 }
 
+/*
+  Splits the given path into it's root and extension.
+  Leading periods in the path are ignored.
+
+  :arg path: A string file name, not necessarily valid.
+  :type path: `string`
+
+  :returns: A tuple of the form ``(root, ext)``.
+  :rtype: `(string, string)`
+
+*/
+
+proc splitExt(path:string): (string, string) {
+  var lead = 0;
+  while lead < path.size && path[lead] == "." {
+    lead += 1;
+  }
+
+  var lastIdx = path[lead..].rfind(".") + lead;
```

Comment:
This seems maybe unnecessarily complicated to me?  Couldn't you just use rfind(".") on the path argument to start?  Then check against if the result is either `-1` or the first index?

Reply:
@lydia-duncan The python documentation specifies that it ignores all leading zeroes.
For e.g. `splitExt("../foo.bar")` should output `(../foo, .bar)`

Reply:
Ah, I see.  So if it was just `splitExt("../foo")`, this would ensure it returns `("../foo", "")` instead of `(".", "./foo")?  Seems like a good test case to me :)

Reply:
Yeah that's right.